File: frontend/src/lib/types/requests.ts
// start request types
export interface StartRequest {
	server_url: string;
	token: string;
	paths: string[];
	cooldown: number;
}

// scan request types
export interface ScanRequest {
	paths: string[];
	server_url: string;
	token: string;
}

// get status response types
export interface StatusResponse {
	is_watching: boolean;
	paths: string[];
	server: string | null;
	cooldown: number;
}

// cached watchlist response types
export interface WatchlistCache {
	paths: string[];
	cooldown: number;
}

File: backend/internal/requests/requests.go
package requests

type StartRequest struct {
	ServerUrl string   `json:"server_url"`
	Token     string   `json:"token"`
	Paths     []string `json:"paths"`
	Cooldown  int      `json:"cooldown"` // seconds; used as debounce
}

type ScanRequest struct {
	ServerUrl string   `json:"server_url"`
	Token     string   `json:"token"`
	Paths     []string `json:"paths"`
}

type ListSectionsRequest struct {
	ServerUrl string `json:"server_url"`
	Token     string `json:"token"`
}
